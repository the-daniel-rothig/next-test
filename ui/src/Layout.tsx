import Next from './next.png';
import { ReactNode } from 'react';
import styled from 'styled-components';

type LayoutProps = {
    children?: ReactNode;
}

const Wrapper = styled.div`
    font-family: Azo-Sans, sans-serif;
    background-color: #000;
    color: #fff;
`;

const Nav = styled.nav`
    display: flex;
    justify-content: start;
    align-items: center;
    padding: 10px 20px;
`;

const Img = styled.img`
    width: 150px;
    margin-right: 20px;
`;

const AppHeading = styled.h1`
    font-size: 2rem;
    font-weight: bold;
`;

const Main = styled.main`
    background-color: #fff;
    color: #000;
    min-height: 200px;
    padding: 20px 80px;
`;

const MainInner = styled.div`
    max-width: 800px;
    margin: auto;
`;

const Footer = styled.footer`
    padding: 20px;
`;

export const Layout = ({children}: LayoutProps) => {
    return (
        <Wrapper>
            <Nav>
                <Img src={Next} />
                <AppHeading>
                    NextTrack
                </AppHeading>
            </Nav>
            <Main>
                <MainInner>
                    {children}
                </MainInner>
            </Main>
            <Footer>
                &copy; 2022
            </Footer>
        </Wrapper>
    );
}
