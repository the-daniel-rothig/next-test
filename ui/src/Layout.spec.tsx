import { render, screen } from '@testing-library/react';
import { Layout } from './Layout';

describe('Layout', () => {
    it('has the expected structure', () => {
        const res = render(<Layout>Hello!</Layout>);
        expect(res.container.firstChild).toMatchSnapshot();
    });

    it('renders the child node', () => {
        const res = render(<Layout><div data-testid="found" /></Layout>);
        res.getByTestId('found');
    })
})