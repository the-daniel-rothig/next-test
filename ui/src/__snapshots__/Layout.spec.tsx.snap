// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout has the expected structure 1`] = `
<div
  class="sc-bdvvtL deJxcu"
>
  <nav
    class="sc-gsDKAQ eBQKJi"
  >
    <img
      class="sc-dkPtRN hWfEOd"
      src="next.png"
    />
    <h1
      class="sc-hKwDye gJASDp"
    >
      NextTrack
    </h1>
  </nav>
  <main
    class="sc-eCImPb xTtml"
  >
    <div
      class="sc-jRQBWg gslwKI"
    >
      Hello!
    </div>
  </main>
  <footer
    class="sc-gKclnd gEzmNI"
  >
    Â© 2022
  </footer>
</div>
`;
